enhancements/cloud-integration/aws/aws-sts-support.md

```diff
@@ -0,0 +1,171 @@
+     ```yaml
+apiVersion: v1
+stringData:
+  credentials: |-
+    [default]
+    role_arn=[ AWS Role ARN created for CredentialsRequest ]
+    web_identity_token_file=/var/run/secrets/openshift/serviceaccount/token
+kind: Secret
+metadata:
+  name: [ CredReq .spec.SecretName ]
+  namespace: [ CredReq .spec.Namespace ]
+type: Opaque
+```
+
+Cluster install workflow becomes:
+
+  1. Create an S3 bucket to publish OIDC keys.
+  1. Generate and publish OIDC keys.
+  1. Create an AWS IAM identity provider.
+  1. Configure their local AWS credentials to use STS.
+  1. `openshift-install create install-config`
+  1. Provide their OIDC keys in a manifest, which will be used to configure the Kube APIServer to be used to sign bound service account keys so they are trusted by AWS.
+  1. Modify `spec.credentialsMode` in the generated install-config.yaml to set CCO to manual mode. (this mechanism already exists)
+  1. Generate secret yaml containing an AWS credentials file which configures the client to use STS, and references to token file mounted in. (see above)
+  1. Include secret yaml definitions generated in the steps above.
+  1. `openshift-install create cluster`
+
+The OpenShift cluster will now install without ever using a long live credential. Kubernetes will handle the rotation of the tokens via the projected service account, which are then trusted by AWS via the OIDC / identity provider trust relationship established by the user manually before installing the cluster.
+
+A [tool has been written](https://github.com/sjenning/sts-preflight) that may be available to help users automate some of the initial AWS setup.
+
+Documentation will be provided on this install process, as well as rotating the OIDC signing keys.
+
+#### Future Phases
+
+In the future we may explore automating the OIDC setup and credentials secrets with CCO, instead of placing it in manual mode. However we are unsure if this will proceed and thus will leave this for a future enhancement or addition to this document.
+
+The `CredentialsRequest` API should eventually be updated to include a reference to the ServiceAccount to configure in the policy, so this information no longer needs to be duplicated and maintained in documentation.
+
+
+### Risks and Mitigations
+
+This implementation does not use the AWS pod identity webhook we began deploying in 4.6, meaning our approach differs from EKS.
+
+Non-CVO managed third-party or OLM operators may be using `CredentialsRequests`. As per any use of CCO manual mode, these requests will not be fulfilled automatically. They will have the flexibility to do this with STS, or traditional credentials, depending on what the operator supports.
+
+## Design Details
+
+### Test Plan
+
+It is likely infeasible to automate testing around a manual CCO mode with non-trivial setup steps. For the most part we will need to rely on manual testing by engineering and QE test plans. Future phases where we may automate more of this could change this.
```

Comment:
Here's a test plan,

### Account Setup
1. Create an AWS IAM OIDC provider in CI account. Make sure this and related resources are skipped from pruning.
2. Create IAM roles for all the operators using the OIDC provider. Make sure these roles are also skipped from pruning.

### Adding another multi-step workflow

3. Add a ipi-conf step to set the credentialMode: Manual
4. Store the Bound Service Account Singning key in CI cluster as secret.
5. Use CI steps to inject this credential to one of the step see https://docs.ci.openshift.org/docs/architecture/step-registry/#injecting-custom-credentials

6. Create a step that adds following items to the $SHARED_DIR/manifests such that the install step will add those files before running create cluster.
   - manifests/cluster-authentication-02-config.yaml (static)
   - tls/bound-service-account-signing-key.key (from the credential)
   - manifests/secret-{* for each operator} (static)
   see https://github.com/openshift/release/blob/ed755e202de5d04dbeb72ec91670f13e9a722022/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L45-L49
   maybe also enable non-manifest folder for tls.

7. This should test OpenShift clusters components using sts for AWS authentication and running the same e2e.

Reply:
Thanks!